frontera/contrib/backends/remote/codecs/json.py

```diff
@@ -3,29 +3,59 @@
 """
 from __future__ import absolute_import
 import json
+import six
 from base64 import b64decode, b64encode
 from frontera.core.codec import BaseDecoder, BaseEncoder
-from w3lib.util import to_unicode, to_native_str
-from frontera.utils.misc import dict_to_unicode, dict_to_bytes
+from w3lib.util import to_unicode, to_bytes
+
+
+def _encode_recursively(obj):
+    """
+    recursively encodes an object to unicode and store its type
```

Comment:
a nitpick: you can't encode to unicode because unicode is not an encoding (at least in Python), you can only decode to unicode